### PR TITLE
Task-50856 : [Active directory sycnhronization] AD group users aren't…

### PR DESCRIPTION
**Pull request by rdenarie**
… synchronized with the eXo group

Before this fix, when an AD group is member of more than one group, he is synchronized at root group level. When a parent of this group is removed (he now have only one parent), he is sync ad children of his parent. But the  information is not propagated to idm database. So, when a user, member of a group like this connect, he is ejected from this group.
This fix allow to detect that a group moves from the root level to a non root level, and propagate the information to the IDM database

### CODE REVIEW

File: component/identity/src/main/java/org/exoplatform/services/organization/idm/GroupDAOImpl.java [hunk 1]
```diff
@@ -152,6 +152,96 @@ public void addChild(Group parent, Group child, boolean broadcast) throws Except
...
+        try {
+            jbidGroupToMove = getIdentitySession().getPersistenceManager()
+                                                        .findGroup(plGroupToMoveName,
+                                                                   orgService.getConfiguration().getGroupType(groupToMove.getParentId()));
+        } catch (Exception e) {
+            // TODO:
+            handleException("Identity operation error: ", e);
+        }
+
+        //if one is miossing => error
```
Comment (ahamdi) [inline component/identity/src/main/java/org/exoplatform/services/organization/idm/GroupDAOImpl.java]:
```suggestion
        //if one is missing => error
```

File: component/identity/src/main/java/org/exoplatform/services/organization/idm/GroupDAOImpl.java [hunk 2]
```diff
@@ -152,6 +152,96 @@ public void addChild(Group parent, Group child, boolean broadcast) throws Except
...
+        if (jbidParentOriginGroup == null) {
+            throw new Exception("Group " + jbidParentOriginGroup + " does not exists");
```
Comment (ahamdi) [inline component/identity/src/main/java/org/exoplatform/services/organization/idm/GroupDAOImpl.java]:
```suggestion
            throw new Exception("Group " + jbidParentOriginGroup + " does not exist");
```

File: component/identity/src/main/java/org/exoplatform/services/organization/idm/GroupDAOImpl.java [hunk 3]
```diff
@@ -152,6 +152,96 @@ public void addChild(Group parent, Group child, boolean broadcast) throws Except
...
+        }
+        if (jbidParentTargetGroup == null) {
+            throw new Exception("Group " + jbidParentOriginGroup + " does not exists");
```
Comment (ahamdi) [inline component/identity/src/main/java/org/exoplatform/services/organization/idm/GroupDAOImpl.java]:
```suggestion
            throw new Exception("Group " + jbidParentOriginGroup + " does not exist");
```

File: component/identity/src/main/java/org/exoplatform/services/organization/idm/GroupDAOImpl.java [hunk 4]
```diff
@@ -152,6 +152,96 @@ public void addChild(Group parent, Group child, boolean broadcast) throws Except
...
+        }
+        if (jbidGroupToMove == null) {
+            throw new Exception("Group " + jbidGroupToMove + " does not exists");
```
Comment (ahamdi) [inline component/identity/src/main/java/org/exoplatform/services/organization/idm/GroupDAOImpl.java]:
```suggestion
            throw new Exception("Group " + jbidGroupToMove + " does not exist");
```

File: component/identity/src/main/java/org/exoplatform/services/organization/idm/GroupDAOImpl.java [hunk 5]
```diff
@@ -152,6 +152,96 @@ public void addChild(Group parent, Group child, boolean broadcast) throws Except
...
+//            jbidOriginGroup = getIdentitySession().getPersistenceManager().findGroup(plParentOriginGroupName,
+//                                                                                     orgService.getConfiguration().getGroupType(originGroup.getParentId()));
+//        } catch (Exception e) {
+//            // TODO:
+//            handleException("Identity operation error: ", e);
+//        }
+//
+//        if (jbidOriginGroup == null) {
+//
+//        }
```
Comment (ahamdi) [inline component/identity/src/main/java/org/exoplatform/services/organization/idm/GroupDAOImpl.java]: COuld you remove this block if it is not needed

Comment (rdenarie) [reply component/identity/src/main/java/org/exoplatform/services/organization/idm/GroupDAOImpl.java]: done

**Comment by rdenarie** [comment]
cc @ahamdi  can you validate to merge

**Review by ahamdi — APPROVED** [review]

**Pull request merged**